playground/src/input-bar/input-bar.js

```diff
@@ -27,6 +34,82 @@ class Input {
     this.input = target.querySelector('input');
     this.submit = target.querySelector('button');
     this.label = target.querySelector('label');
+
+    if (config.autocomplete) {
+      this.createAutocomplete(config.autocomplete.options);
+    }
+
+    this.eventBus = new EventBus();
+  }
+
+  subscribe(channel, observer) {
+    this.eventBus.subscribe(channel, observer);
+  }
+
+  createAutocomplete(options) {
+    this.autocomplete = this.container.querySelector('.input-bar-autocomplete');
+    this.autocomplete
+      .querySelector('.input-bar-autocomplete-header-close')
+      .addEventListener('click', () => {
+        this.toggleAutocomplete(false);
+      });
+
+    this.input.addEventListener(
+      'keyup',
+      this.filterAutocompleteOptions.bind(this)
+    );
+    this.input.addEventListener('focus', () => {
+      this.toggleAutocomplete(true);
+    });
+
+    if (options.length) {
+      this.updateAutocompleteOptions(options);
+    }
+  }
+
+  updateAutocompleteOptions(options) {
+    this.autocompleteOptions = options;
+    this.renderAutocompleteList(options);
+  }
+
+  renderAutocompleteList(options) {
```

Comment:
As this is triggered for every `keyup` it would be nice to optimize this: wouldn't it work to simply toggle visibility for filtered items?

Reply:
so ... no?